src/view/index.js

```diff
@@ -36,18 +36,33 @@ export default class View {
         let self = this;
         let { dispatch } = this.store;
 
+        // incase we're currently running a job, stop the old one
+        this.stop();
```

Comment:
Theres a race condition lurking in here, if you `run()` two times in a row, on the second `run()`, `this._jobSocket` will still be `null` so you'll end up assigning to `self._jobSocket` twice without calling `stop()` in between.